### PR TITLE
Add "Remember last folder in file pickers" setting, enabled by default

### PR DESCRIPTION
**Pull request by Keelhauled**

### CODE REVIEW

**Comment by Keelhauled** [comment]
Can't test because no winblows

File: src/Shared.Core/KoikatuAPIBase.cs [hunk 1]
```diff
@@ -84,6 +85,7 @@ private void BaseAwake()
             Logger = base.Logger;
 
             EnableDebugLoggingSetting = Config.Bind("Debug", "Show debug messages", false, "Enables display of additional log messages when certain events are triggered within KKAPI. Useful for plugin devs to understand when controller messages are fired. Changes take effect after game restart.");
+            RememberFilePickerFolder = Config.Bind("General", "Remember file picker folder", true, "Setting only applies to plugins that use the file picker through this API.");
```
Comment (ManlyMarco) [inline src/Shared.Core/KoikatuAPIBase.cs:88]:
```suggestion
            RememberFilePickerFolder = Config.Bind("General", "Remember last folder in file pickers", true, "If true, file picker dialogs will remember the last opened folder and open already inside of it. If false, the dialogs will always open in the default folder.\n\nWarning: This setting only applies to plugins that use the file picker through this API.");
```

File: src/Shared.Core/Utilities/OpenFileDialog.cs [hunk 2]
```diff
@@ -66,7 +73,10 @@ public static string[] ShowDialog(string title, string initialDir, string filter
             ofn.filter = filter.Replace("|", "\0") + "\0";
             ofn.fileTitle = new String(new char[MAX_FILE_LENGTH]);
             ofn.maxFileTitle = ofn.fileTitle.Length;
-            ofn.initialDir = initialDir;
+            if (LastOpenedPaths.TryGetValue(initialDir, out string lastOpenedPath) && KoikatuAPI.RememberFilePickerFolder.Value)
+                ofn.initialDir = lastOpenedPath;
+            else
+                ofn.initialDir = initialDir;
```
Comment (ManlyMarco) [inline src/Shared.Core/Utilities/OpenFileDialog.cs:79]: Probably better to just check if lastOpenedPath is null or empty, and check the setting first.

File: src/Shared.Core/Utilities/OpenFileDialog.cs [hunk 3]
```diff
@@ -103,6 +114,7 @@ public static string[] ShowDialog(string title, string initialDir, string filter
 
             run = false;
             Environment.CurrentDirectory = currentWorkingDirectory;
+
```
Comment (ManlyMarco) [inline src/Shared.Core/Utilities/OpenFileDialog.cs:117]:
```suggestion
```

File: src/Shared.Core/Utilities/OpenFileDialog.cs [hunk 4]
```diff
@@ -89,6 +99,7 @@ public static string[] ShowDialog(string title, string initialDir, string filter
             // Save and restore working directory since GetOpenFileName changes it for no good reason
             var currentWorkingDirectory = Environment.CurrentDirectory;
             var run = true;
+
```
Comment (ManlyMarco) [inline src/Shared.Core/Utilities/OpenFileDialog.cs:102]:
```suggestion
```

**Review by ManlyMarco — APPROVED** [review]

**Pull request merged**